x-pack/test/api_integration/apis/ml/saved_objects/sync_trained_models.ts

```diff
@@ -0,0 +1,205 @@
+  const modelIdSpace3 = 'dfa3_classification_model_n_0';
+
+  async function runSyncRequest(user: USER, expectedStatusCode: number) {
+    const { body, status } = await supertest
+      .get(`/s/${idSpace1}/api/ml/saved_objects/sync`)
+      .auth(user, ml.securityCommon.getPasswordForUser(user))
+      .set(COMMON_REQUEST_HEADERS);
+    ml.api.assertResponseStatusCode(expectedStatusCode, status, body);
+
+    return body;
+  }
+
+  async function runSyncCheckRequest(user: USER, expectedStatusCode: number) {
+    const { body, status } = await supertest
+      .post(`/s/${idSpace1}/api/ml/saved_objects/sync_check`)
+      .auth(user, ml.securityCommon.getPasswordForUser(user))
+      .set(COMMON_REQUEST_HEADERS)
+      .send({ mlSavedObjectType: 'trained-model' });
+    ml.api.assertResponseStatusCode(expectedStatusCode, status, body);
+
+    return body;
+  }
+
+  function getTestModel(id: string, modelType: ModelType, jobId?: string) {
+    return {
+      model_id: id,
+      body: {
+        compressed_definition: ml.api.getCompressedModelDefinition(modelType),
+        inference_config: {
+          [modelType]: {},
+        },
+        ...(jobId ? { metadata: { analytics_config: { id: jobId } } } : {}),
+        input: {
+          field_names: ['common_field'],
+        },
+      } as PutTrainedModelConfig,
+    };
+  }
+
+  describe('GET saved_objects/sync', () => {
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/ihp_outlier');
+    });
+
+    beforeEach(async () => {
+      await ml.api.initSavedObjects();
+      await spacesService.create({ id: idSpace1, name: 'space_one', disabledFeatures: [] });
+      await spacesService.create({ id: idSpace2, name: 'space_two', disabledFeatures: [] });
+      await ml.testResources.setKibanaTimeZoneToUTC();
+    });
+
+    afterEach(async () => {
+      await spacesService.delete(idSpace1);
+      await spacesService.delete(idSpace2);
+      await ml.api.cleanMlIndices();
+      await ml.testResources.cleanMLSavedObjects();
+    });
+
+    it('should sync trained model saved objects', async () => {
+      // check to see if a sync is needed
+      const syncNeeded = await runSyncCheckRequest(USER.ML_POWERUSER_ALL_SPACES, 200);
+      expect(syncNeeded.result).to.eql(false, 'sync should not be needed');
+
+      await ml.api.createDataFrameAnalyticsJob(
+        ml.commonConfig.getDFAIhpOutlierDetectionJobConfig(dfaJobId1),
+        idSpace1
+      );
+      await ml.api.updateJobSpaces(dfaJobId1, 'data-frame-analytics', [idSpace2], [], idSpace1);
+      await ml.api.assertJobSpaces(dfaJobId1, 'data-frame-analytics', [idSpace1, idSpace2]);
+
+      // create model in es, so saved object creation is needed
+      const model1 = getTestModel(modelIdSpace1, 'classification', dfaJobId1);
+      await ml.api.createTrainedModelES(model1.model_id, model1.body);
+
+      // create model via kibana, so saved object creation is no needed
```

Comment:
Nit - should be `not needed`

Reply:
https://github.com/elastic/kibana/pull/130115/commits/a65265454a4be24e54f6d412652fff6638936047